pkg/services/ngalert/state/testing.go

```diff
@@ -35,3 +35,9 @@ func (f *FakeInstanceStore) DeleteAlertInstance(_ context.Context, _ int64, _, _
 func (f *FakeInstanceStore) DeleteAlertInstancesByRule(ctx context.Context, key models.AlertRuleKey) error {
 	return nil
 }
+
+type FakeRuleReader struct{}
+
+func (f *FakeRuleReader) ListAlertRules(_ context.Context, q *models.ListAlertRulesQuery) error {
+	return nil
```

Comment:
The tests don't actually cover any calls to the fake, so this was all that was needed to get current tests to pass.

Extending this in the future if we need is trivial though, we'd just extend `FakeRuleReader` with a slice of rules to return. The minimal interface keeps this small and bounded, and prevents this fake from spiraling out of control like the other one.